File: 0682/Sesión 05 - lib C#/Publicaciones_v00/Publicaciones_v00/Publicacion.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Globalization;
using System.Linq;
using System.Runtime.Remoting.Metadata.W3cXsd2001;
using System.Text;
using System.Threading.Tasks;

namespace Publicaciones_v00
{

    public abstract class Publicacion
    {
        
        public static int orden_presentacion = 1;
        protected string nombre;
        protected ArrayList autores;
        protected int cantidad_autores;
        protected int año;

        public Publicacion(string nombre, string nombre_completo_autor, int año)
        {
            this.nombre = nombre;
            this.autores = new ArrayList();
            Autor autor = new Autor(nombre_completo_autor);
            this.autores.Add(autor);
            this.cantidad_autores = 1;
            this.año = año;
        }

        public Publicacion(string nombre, string[] nombres_completos_autores, int año)
        {
            this.nombre = nombre;
            this.autores = new ArrayList();
            for (int i = 0; i < nombres_completos_autores.Length; i++)
            {
                Autor autor = new Autor(nombres_completos_autores[i]);
                this.autores.Add(autor);
            }
            this.cantidad_autores = autores.Count;
            this.año = año;
        }

        public void Referenciar()
        {
            Console.WriteLine(this);
        }
    }
}

File: 0683/Sesión 04 - lib java/Publicaciones_v02/src/main/java/pe/edu/pucp/inf30/sesiones/publicaciones02/publicaciones/Publicacion.java
package pe.edu.pucp.inf30.sesiones.publicaciones02.publicaciones;

import java.util.ArrayList;

public abstract class Publicacion {

    public static Integer orden_presentacion = 1;
    protected String nombre;
    protected ArrayList<Autor> autores;
    protected Integer cantidad_autores;
    protected Integer año;

    public Publicacion(String nombre, String nombre_completo_autor, Integer año) {
        this.nombre = nombre;
        this.autores = new ArrayList<>();
        Autor autor = new Autor(nombre_completo_autor);
        this.autores.add(autor);
        this.cantidad_autores = 1;
        this.año = año;
    }

    public Publicacion(String nombre, String nombres_completos_autores[], Integer año) {
        this.nombre = nombre;
        this.autores = new ArrayList<>();
        for (String nombres_completos_autor_loop : nombres_completos_autores) {
            Autor autor = new Autor(nombres_completos_autor_loop);
            this.autores.add(autor);
        }
        this.cantidad_autores = this.autores.size();
        this.año = año;
    }
       
    public void referenciar(){
        System.out.println(this);
    }
}
